plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/HudiPageSourceProvider.java

```diff
@@ -84,16 +87,19 @@
 public class HudiPageSourceProvider
         implements ConnectorPageSourceProvider
 {
+    private final TrinoFileSystemFactory fileSystemFactory;
     private final HdfsEnvironment hdfsEnvironment;
     private final Map<HoodieFileFormat, HudiPageSourceCreator> pageSourceBuilderMap;
 
     @Inject
     public HudiPageSourceProvider(
+            TrinoFileSystemFactory fileSystemFactory,
             HdfsEnvironment hdfsEnvironment,
             FileFormatDataSourceStats fileFormatDataSourceStats,
             ParquetReaderConfig parquetReaderConfig,
             HudiConfig hudiConfig)
     {
+        this.fileSystemFactory = requireNonNull(fileSystemFactory, "fileSystemFactory is null");
         this.hdfsEnvironment = requireNonNull(hdfsEnvironment, "hdfsEnvironment is null");
```

Comment:
Do we still need to pass `HdfsEnvironment` to `HudiParquetPageSourceCreator`, or can that be removed now?

Reply:
Both `Configuration` and `HdfsEvironment` can be removed now.